### PR TITLE
DDCNLS-3906 | Updated date validation

### PR DESCRIPTION
**Pull request by JackTattersall**

### CODE REVIEW

File: app/forms/DateOfLeavingForm.scala [hunk 1]
```diff
@@ -64,10 +68,10 @@ object DateOfLeavingForm {
         "month" -> optional(text),
         "year" -> optional(text)
       )(GmpDate.apply)(GmpDate.unapply)
-        .verifying(Messages("gmp.error.date.nonnumber"), x => checkForNumber(x.day) && checkForNumber(x.month) && checkForNumber(x.year))
-        .verifying(Messages("gmp.error.day.invalid"), x => checkDayRange(x.day))
-        .verifying(Messages("gmp.error.month.invalid"), x => checkMonthRange(x.month))
-        .verifying(Messages("gmp.error.year.invalid.format"), x => checkYearLength(x.year))
+//        .verifying(Messages("gmp.error.date.nonnumber"), x => checkForNumber(x.day) && checkForNumber(x.month) && checkForNumber(x.year))
```
Comment (shridhaarchippaa) [inline app/forms/DateOfLeavingForm.scala]: Please remove the commented code.

File: app/models/GMPDate.scala [hunk 2]
```diff
@@ -39,6 +39,15 @@ case class GmpDate(day: Option[String], month: Option[String], year: Option[Stri
       false
   }
 
+  def isBefore05042046: Boolean = {
+    if (getAsLocalDate.isDefined) {
+      val thatDate = new LocalDate(2046, 4, 5)
+      getAsLocalDate.get.isBefore(thatDate)
```
Comment (shridhaarchippaa) [inline app/models/GMPDate.scala]: Generally tax year ends on 5 April and starts from 6 April. Not sure whether that is relevant here.

File: test/models/GmpDateSpec.scala [hunk 3]
```diff
@@ -49,6 +49,28 @@ class GmpDateSpec extends PlaySpec with MockitoSugar{
...
 
+  "isBeforer01012100" must {
+
+    "return true when before 05042046" in {
+
+      val before = GmpDate(Some("4"),Some("4"),Some("2046"))
+      before.isBefore05042046 must be(true)
+    }
+
+    "return false when on or after 01012099" in {
```
Comment (shridhaarchippaa) [inline test/models/GmpDateSpec.scala]: title is confusing?

Comment (periklas) [reply test/models/GmpDateSpec.scala]: This has now been addressed!

**Pull request merged**